src/core/controls.js

```diff
@@ -0,0 +1,177 @@
+var component = require('./component');
+var systems = require('./system');
+var utils = require('../utils/');
+
+// Keep track of registered controls.
+var controls = module.exports.controls = {
+  movementControls: {},
+  rotationControls: {}
+};
+
+/**
+ * Initializes the A-Frame controls system.
+ */
+systems.registerSystem('controls', {
+  init: function () {
+    // Names of enabled movement/rotation controls, in order of precedence.
+    this.enabledMovementControls = [];
+    this.enabledRotationControls = [];
+
+    // Controls components registered to update object position/rotation.
+    this.movementControls = controls.movementControls;
+    this.rotationControls = controls.rotationControls;
+  },
+
+  /**
+   * Enables the given component names, and sets the given order as their precedence.
+   * @param {Array<string>} controls
+   */
+  enableMovementControls: function (controls) {
+    this.enabledMovementControls = controls;
+  },
+
+  /**
+   * Enables the given component names, and sets the given order as their precedence.
+   * @param {Array<string>} controls
+   */
+  enableRotationControls: function (controls) {
+    this.enabledRotationControls = controls;
+  },
+
+  /**
```

Comment:
Yeah, makes sense. It would be nice to support multiple gamepads for example.

I've moved this logic back into the controls component, so in theory we can support several of them. Also used the system's registry to detect invalid controls and inject missing ones on load.